lib/fork.js

```diff
@@ -2,7 +2,11 @@ const childProcess = require('child_process')
     , childModule  = require.resolve('./child/index')
 
 function fork (forkModule) {
-  var child = childProcess.fork(childModule, {
+  // suppress --debug flags while preserving others (like --harmony)
+  var filteredArgs = process.execArgv.filter(function (v) {
+        return v == '--debug'
+      })
+    , child        = childProcess.fork(childModule, { execArgv: filteredArgs }, {
```

Comment:
I'm not sure what you mean. I don't want to dive down a rabbit hole here,
just thought the blocker was the style. Can we chat on gitter if you think
we can resolve this relatively quickly?

On Tue, Sep 29, 2015 at 11:24 AM Rod Vagg <notifications@github.com> wrote:

> In lib/fork.js
> <https://github.com/rvagg/node-worker-farm/pull/34#discussion_r40686102>:
>
> > @@ -2,7 +2,11 @@ const childProcess = require('child_process')
> >      , childModule  = require.resolve('./child/index')
> >
> >  function fork (forkModule) {
> > -  var child = childProcess.fork(childModule, {
> > +  // suppress --debug flags while preserving others (like --harmony)
> > +  var filteredArgs = process.execArgv.filter(
>
> function (v) {
> > +        return v == '--debug'
> > +      })
> > +    , child        = childProcess.fork(childModule, { execArgv: filteredArgs }, {
>
> I think that either execArgv should be in the options list specified as
> the last argument or you should just supply filteredArgs as the second
> argument (plain, as an array). The former is probably the best call, I
> think, tho you might want to test!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rvagg/node-worker-farm/pull/34/files#r40686102>.
>